src/dune_rules/lib_rules.ml

```diff
@@ -483,6 +483,27 @@ let setup_build_archives (lib : Library.t) ~top_sorted_modules ~cctx ~expander ~
     (fun () -> build_shared ~native_archives ~sctx lib ~dir ~flags)
 ;;
 
+let parameterize_root_module ~parameter ~root_module_name modules =
```

Comment:
Instead of storing this information in the module, how about just add a field to the compilation context designating which module is implementing a parameter. Should be easier to understand and more efficient.